### PR TITLE
Add EquinixMetal config api types

### PR DESCRIPTION
**Pull request by detiber**
This PR contains changes needed to facilitate Equinix Metal support for machine-api-operator

### CODE REVIEW

File: config/v1/types_infrastructure.go [hunk 1]
```diff
@@ -116,6 +116,9 @@ const (
 
 	// KubevirtPlatformType represents KubeVirt/Openshift Virtualization infrastructure.
 	KubevirtPlatformType PlatformType = "KubeVirt"
+
+	// EquinixMetalPlatformType represents Equinix Metal infrastructure.
+	EquinixMetalPlatformType PlatformType = "EquinixMetal"
```
Comment (elmiko) [inline config/v1/types_infrastructure.go:121]: does the `+kubebuilder:validation` comment on the top of this structure need to be regenerated to include `EquinixMetal`?

**Review by elmiko — COMMENTED** [review]
this seems mostly reasonable to me, i just have a question about one of the kubebuilder comments.

File: config/v1/types_infrastructure.go (hunk 1 above)
Comment (sttts) [reply config/v1/types_infrastructure.go:121]: yes

File: config/v1/types_infrastructure.go [hunk 2]
```diff
@@ -234,6 +241,10 @@ type PlatformStatus struct {
 	// Kubevirt contains settings specific to the kubevirt infrastructure provider.
 	// +optional
 	Kubevirt *KubevirtPlatformStatus `json:"kubevirt,omitempty"`
+
+	// EquinixMetal contains settings specific to the Equinix Metal infrastructure provider.
+	// +optional
+	EquinixMetal *EquinixMetalPlatformStatus `json:"equinixmetal,omitempty"`
```
Comment (sttts) [inline config/v1/types_infrastructure.go:247]: equinixMetal ?

File: config/v1/types_infrastructure.go (hunk 1 above)
Comment (detiber) [reply config/v1/types_infrastructure.go:121]: Updated, thanks!

File: config/v1/types_infrastructure.go (hunk 2 above)
Comment (detiber) [reply config/v1/types_infrastructure.go]: Done, I was following the convention of the others, but agree this is preferred.

**Review by elmiko — APPROVED** [review]
thanks Jason, that was the only thing that looked odd to me.
/lgtm

**Comment by sttts** [comment]
/approve
/lgtm

/hold
until master reopens.

**Comment by detiber** [comment]
Rebased to resolve conflict

**Review by elmiko — APPROVED** [review]
thanks Jason!
/lgtm

**Comment by crawford** [comment]
Development branch has re-opened.

/hold cancel

**Pull request merged**